src/StateSmithTest/Output/Sim/diagrams/BeadSorter.sim.html

```diff
@@ -3159,8 +3159,21 @@
         // choose to implement a tracer for debugging purposes.
         sm.tracer = {
             enterState: (mermaidName) => {
-                document.querySelector('g[data-id=' + mermaidName + ']')?.classList.add('active');
-                sm.tracer.log("➡️ Entered " + mermaidName);
+                var e = document.querySelector('g[data-id=' + mermaidName + ']');
+                e?.classList.add('active');
+                var bounds = e?.getBoundingClientRect();
+                if(bounds?.x<0 || bounds?.y<0) {
+                    var x = Math.max(0, -bounds.x + 20);
+                    var y = Math.max(0, -bounds.y + 20);
+                    window.panZoom.panBy({x: x, y: y});
```

Comment:
# 🎉 🎉 🎉
This works really well! Nicely done :)  I was a bit worried it might be jarring, but I think it's great!

It's so good that we could push the transition animation work further if we want to focus on something else.